.github/workflows/build.yaml

```diff
@@ -89,39 +72,60 @@ jobs:
       uses: actions/checkout@v3
       with:
         fetch-depth: 1
-    - name: Installing dependencies
-      run: sh ./.github/scripts/install_deps.sh
-    - name: CMake
-      run: |
-        # Prepare build dir
-        echo "Will use $(nproc) cores."
-        mkdir build
-        pushd build
-        cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ..
-        popd
-    - name: Restore cache
-      uses: actions/cache/restore@v3
-      with:
-        path: build/clang-tidy-cache
-        key: clang-tidy-cache-${{ github.sha }}
-        restore-keys: |
-          clang-tidy-cache-
-    - name: Check
-      # To run a single check, call:
-      # python3 ../utils/run-clang-tidy.py -checks=-*,my-check-prefix* | tee ../clang-tidy.log
-      run: |
-        # run-clang-tidy.py
-        pushd build
-        python3 ../utils/run-clang-tidy.py -cache | tee ../clang-tidy.log
-        popd
-    - name: Save cache
-      # We want to save the cache even if the report fails because of failed files
-      uses: actions/cache/save@v3
+    - name: Check for changed files
+      uses: dorny/paths-filter@v2
+      id: changed_files
       with:
-        path: build/clang-tidy-cache
-        key: clang-tidy-cache-${{ github.sha }}
-    - name: Report
-      run: utils/check_clang_tidy_results.py clang-tidy.log
+        # groups to check for changed files (as string, yaml anchors are resolved by the tool)
+        filters: |
+          clang_tidy:
+            - 'src/**'
+            - 'utils/check_clang_tidy_results.py'
+            - 'utils/run-clang-tidy.py'
+          compile: &compile
+            - 'CMakeLists.txt'
+            - 'cmake/**'
+            - 'src/**'
```

Comment:
Why would the installers and test run when build.yaml changes? It is unlikely that a change is related to this jobs. (The file still has many jobs inside.) In case paths for filters are changed in build.yaml, this triggers a change usually because the file was changed as well.
No job checks for build.yaml currently.

Reply:
https://github.com/widelands/widelands/pull/6028#issuecomment-1817283718

Reply:
Adding a dependency doesn't necessarily mean that that dependency is changed in the given PR.

But OK, if there are no actual changes in the code, and all build instructions are in the separated sub-workflows, then indeed, running them doesn't give much value, it only wastes time. The main yaml is validated anyway, even if it just concludes that everything can be skipped.